src/FsAutoComplete.Core/CompilerServiceInterface.fs

```diff
@@ -453,10 +453,28 @@ type FSharpCompilerServiceChecker(backgroundServiceEnabled) =
 
   member __.GetProjectOptionsFromScript(file, source) = async {
     let targetFramework = NETFrameworkInfoProvider.latestInstalledNETVersion ()
-
     let! projOptions = fsxBinder.GetProjectOptionsFromScriptBy(targetFramework, file, source)
 
-    return projOptions
+    let config =
+      Fake.Runtime.FakeRuntime.createConfigSimple
+        Fake.Runtime.Trace.VerboseLevel.Verbose [] file [] true false
+    let prepared = Fake.Runtime.FakeRuntime.prepareFakeScript config
+    let isFakeScript = prepared.DependencyType <> Fake.Runtime.FakeRuntime.PreparedDependencyType.DefaultDependencies
+    if not isFakeScript then return projOptions
```

Comment:
Until this `isFakeScript` check it should be pretty fast. `prepareFakeScript` basically does tokenize the script, which I feel is acceptable.
in the `fake` branch there is all the logic for downloading NuGet packages and resolving the proper arguments...